### ISSUE TITLE
[NEATO] Exception at NeatoAccountDiscoveryService 

### ISSUE THREAD

**Issue by jensflorian** [openhab/openhab-addons#8930]
## Expected Behavior
Neato Botvac D7 should be detected and added to Inbox in Paper UI.

## Current Behavior
Neato BotVac D7 installed in Neato App and is working.
Neato Account in OpenHab Paper UI is online.
In Paper UI use Inbox ->Neato Binding -> Search

No Botvac is added. In console following output is generated:

```
[2020-11-01 16:50:23.822 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 
java.lang.NullPointerException: null
	at org.eclipse.smarthome.core.thing.ThingUID.<init>(ThingUID.java:54) ~[?:?]
	at org.openhab.binding.neato.internal.discovery.NeatoAccountDiscoveryService.addThing(NeatoAccountDiscoveryService.java:90) ~[?:?]
	at org.openhab.binding.neato.internal.discovery.NeatoAccountDiscoveryService.findRobots(NeatoAccountDiscoveryService.java:59) ~[?:?]
	at org.openhab.binding.neato.internal.discovery.NeatoAccountDiscoveryService.lambda$0(NeatoAccountDiscoveryService.java:73) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

## Possible Solution
Add BotVac manually, however I do not have the "Secret" ID for my BotVac D7.

## Context
This problem was first noticed in OpenHab 2.5.9 and persists in 2.5.10 - it was working previously in June 2019 (OpenHab 2.5.4?)

## Your Environment
Latest OpenHabian with OpenHab 2.5.10 on Raspberry Pi 4.

**Comment by MFI-Net** [comment]
I have the same issue, but don't understand what to do to fix it. 
My system: openhab - 2.5.10-1  an PI4
Can anybody help me.

Best regards Martin

**Comment by McMornan** [comment]
Well, I have the same issue in 3.00 release version:

22:28:14.242 [WARN ] [ommon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
java.lang.NullPointerException: null
        at org.openhab.core.thing.ThingUID.<init>(ThingUID.java:58) ~[?:?]
        at org.openhab.binding.neato.internal.discovery.NeatoAccountDiscoveryService.addThing(NeatoAccountDiscoveryService.java:95) ~[?:?]
        at org.openhab.binding.neato.internal.discovery.NeatoAccountDiscoveryService.findRobots(NeatoAccountDiscoveryService.java:59) ~[?:?]
        at org.openhab.binding.neato.internal.discovery.NeatoAccountDiscoveryService.lambda$0(NeatoAccountDiscoveryService.java:73) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]

**Comment by MFI-Net** [comment]
Any news on this topic??  Can anybody help?
Best regards
Martin

**Comment by McMornan** [comment]
yea, fixed since 3.1m1
